### PR TITLE
Script for Student "Health Check"

### LINKED ISSUE #521: [Feature] Health check of system setup for students
### Use-cases

Students especially in the GSC track are having several issues caused by a insufficient setup of the system (wrong version combinations of Azure Functions Core and Node.js, no Node.js, outdated Nodes.js etc).

This is quite hard to figure out via Piazza and may be a very frustrating experience when starting the serverless journey

### Proposal

Provide shell scripts (I expect three depending on the OS namely Windows, MacOS and Linux) or an executeable to allow a check for:

- Is Node.js installed (yes/no)
- Are Azure Function Core Tools installed (yes/no)
- Are the versions of Azure Function Core Tools and Node.js compatible

This should include hints what to do in case soemthing is missing or incomatible

### References

na

### PR DESCRIPTION
**Pull request by emsesc**
## Changes made
* Wrote script that checks node + core tools versions
* Provides suggestions for fixing

@lechnerc77 I believe this script is compatible with MacOS, Windows, and Linux. Students can run the bash script in Git Bash since they'll have that installed in order to work with Git if they have a Windows system.

Closes #521

### CODE REVIEW

File: checkup.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+echo "Your Azure Functions Core Tools version is $funcv."
+
+echo "--------------------------------";
+echo "Checking if node is installed...";
+echo "--------------------------------";
+if which node > /dev/null
+then
+    echo "Node is installed, skipping...";
+else
+    echo "Node is not installed.";
```
Comment (ganning127) [inline checkup.sh:16]: Would students need to restart their terminal to see that node is infact installed? I know that sometimes that needs to happen. Maybe add a line saying "If you've installed `node` but this script is still saying you haven't, please restart your terminal and run the script again"

**Review by ganning127 — CHANGES_REQUESTED** [review]

**Comment by lechnerc77** [comment]
@emsesc had a first run on Linux and Windows (via Git bash). There are some issues, I will enter them as review comments tomorrow.

File: checkup.sh (hunk 1 above)
Comment (lechnerc77) [reply checkup.sh:16]: We should put that into the output at the very beginning

File: checkup.sh [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+fi
+
+echo "\n"
+echo "---------------------------------------"
+echo "Checking if versions are compatible..."
+echo "---------------------------------------"
+if [[ (${nodev:1:2} == "16" || ${nodev:1:2} == "14") && (${funcv:0:1} == "4") ]]
+then
+    echo "Compatible versions! You are good to go.";
+elif [[ (${nodev:1:2} == "14" || ${nodev:1:2} == "12" || ${nodev:1:2} == "10") && (${funcv:0:1} == "3") ]]
```
Comment (lechnerc77) [inline checkup.sh:48]: Node.js version 10 and 12 are out of maintenance. They should be excluded; maybe a message might make sense when checking the Node version that those are not supported

**Review by lechnerc77 — CHANGES_REQUESTED** [review]

**Comment by emsesc** [comment]
Thanks for the suggestions @lechnerc77 and @ganning127 ! I've implemented them all.

File: checkup.sh [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+#!/bin/bash
+nodev=$(node -v);
+funcv=$(func -v);
+
+echo "🛑 Note: If you've installed node and Azure Functions Core Tools but this script is still saying you haven't, please restart your terminal and run the script again."
+echo "\n"
+echo "--------------------------------";
+echo "Checking if node is installed...";
+echo "--------------------------------";
+if [[ $(node -v) ]]
```
Comment (lechnerc77) [inline checkup.sh:11]: I think we can use `${nodev}` and `${funcv}` in the IF-clauses instead of calling `node-v` again.

Same on line 34

**Review by lechnerc77 — CHANGES_REQUESTED** [review]
One small addition to get rid of another command

File: checkup.sh (hunk 3 above)
Comment (lechnerc77) [reply checkup.sh:11]: @emsesc Overall looks really good now - also works in Git Bash as expected

**Review by lechnerc77 — APPROVED** [review]

**Review by ganning127 — APPROVED** [review]

**Pull request merged**